include/swift/AST/PrintOptions.h

```diff
@@ -294,6 +275,9 @@ struct PrintOptions {
   /// Whether to print the extensions from conforming protocols.
   bool PrintExtensionFromConformingProtocols = false;
 
+  std::shared_ptr<ShouldPrintChecker> CurrentPrintabilityChecker =
```

Comment:
Just `ShouldPrintChecker *` is fine—if it's `nullptr`, you get the default behavior. Whenever someone wants to customize it, they can stack-allocate their checker and pass in the address.